core/appmodule/v2/genesis.go

```diff
@@ -0,0 +1,14 @@
+package appmodule
+
+import (
+	"context"
+)
+
+// HasGenesis defines a custom genesis handling API implementation.
```

Comment:
What are we going to do about genesis in core v1?
I feel like we should delete if when there is consensus about the api, as we never pushed people to use it.